File: hw/nase/u2/hw14 DBMongoandGUI/Python/model/mother_model.py
from model.mongo_connection import MongoConnection

class MotherModel:
    def __init__(self):
        self.db = MongoConnection.get_database()
        self.collection = self.db["Mother"]

    def save_mother(self, mother_data, baby_data):
        mother_id = mother_data["id"]
        document = {
            "firstName": mother_data["firstName"],
            "lastName": mother_data["lastName"],
            "id": mother_id,
            "birthDate": mother_data["birthDate"],
            "weight": mother_data["weight"],
            "height": mother_data["height"],
            "babies": [baby_data] 
        }
        return self.collection.replace_one({"id": mother_id}, document, upsert=True)

File: hw/nase/u2/hw14 DBMongoandGUI/JavaScript/model/motherModel.js
const MongoConnection = require('./mongoConnection');

class MotherModel {
    constructor() {
        this.db = MongoConnection.getDatabase();
        this.collection = this.db.collection("Mother");
    }

    async saveMother(motherData, babyData) {
        const motherId = motherData.id;
        const document = {
            firstName: motherData.firstName,
            lastName: motherData.lastName,
            id: motherId,
            birthDate: motherData.birthDate,
            weight: parseFloat(motherData.weight),
            height: parseFloat(motherData.height),
            babies: [babyData] 
        };
        return await this.collection.replaceOne({ id: motherId }, document, { upsert: true });
    }

    async getMother(motherId) {
        return await this.collection.findOne({ id: motherId });
    }
}

module.exports = MotherModel;